### PR TITLE
rest-api: move config modify rest api handler into rest-api cell

### PR DESCRIPTION
**Pull request by mhofstetter**
Currently, the REST API handler that handles config modifications is implemented in the Daemon struct.

This commit moves the handler and all it's dependencies into the rest api cell.

### CODE REVIEW

**Review by joamaki — APPROVED** [review]

File: daemon/restapi/config.go [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+	changes += option.Config.Opts.ApplyValidated(om, h.changedOption, nil)
+	h.endpointManager.OverrideEndpointOpts(om)
+
+	h.logger.WithField("count", changes).Debug("Applied changes to daemon's configuration")
+	option.Config.ConfigPatchMutex.Unlock()
+
+	if changes > 0 {
+		// Only recompile if configuration has changed.
+		h.logger.Debug("daemon configuration has changed; recompiling base programs")
+		if err := h.datapath.Orchestrator().Reinitialize(context.TODO()); err != nil {
```
Comment (mhofstetter) [inline daemon/restapi/config.go]: TODO: replace `context.TODO()` with a context that spans the lifecycle of the cell (~ agent). Previously this was using `Daemon.ctx`.

**Comment by mhofstetter** [comment]
rebased to `main` to resolve conflicts.

File: daemon/restapi/config.go (hunk 1 above)
Comment (mhofstetter) [reply daemon/restapi/config.go]: fixed. the event handler now also contains a context as field (analogous to the daemon). Even though not the nicest thing, i'd like to keep the logic the same in this PR.

**Comment by mhofstetter** [comment]
rebased on top of `main` to resolve conflicts

**Review by joamaki — APPROVED** [review]

**Review by derailed — APPROVED** [review]
@mhofstetter Nice work Marco!

**Comment by mhofstetter** [comment]
Rebased on top of `main` to resolve conflicts with https://github.com/cilium/cilium/pull/33190.

@ldelossa Gentle ping :)

**Comment by mhofstetter** [comment]
removing review-request for @ldelossa (PTO) - sig-agent is already covered by the review of @joamaki anyway.

**Pull request merged**